src/Symfony/Component/Messenger/EventListener/SendFailedMessageToFailureTransportListener.php

```diff
@@ -27,12 +26,12 @@
  */
 class SendFailedMessageToFailureTransportListener implements EventSubscriberInterface
 {
-    private $failureSender;
+    private $failureSenders;
     private $logger;
 
-    public function __construct(SenderInterface $failureSender, LoggerInterface $logger = null)
+    public function __construct(array $failureSenders, LoggerInterface $logger = null)
```

Comment:
Instead of being an `array`, can you use a service locator instead?

Reply:
`$failureSenders` is an array with the transport name and its failure transport.
The service locator doesn't hold this logic.

Reply:
I understand now the service locator concept 😄thanks!